app/controllers/files/browse_controller.rb

```diff
@@ -10,4 +11,33 @@ def browse
     # load files
     @file = @identity.browse(path)
   end
+
+
+  def upload
+
+    uploader = LocalFileUploader.new
+
+
+    uploader.store!(params[:bruse_file][:file])
+
+
+    file = BruseFile.new
+
+    file.foreign_ref = uploader.file.file
+    file.name = uploader.filename
+    file.filetype = uploader.content_type
+    #temporary solution
+    file.identity = current_user.identities.first
```

Comment:
I agree. (In a coming PR there will be a `User` method named `local_identity`, which could be useful here.)